### PR TITLE
LPS-181694 | Publish Api Application

### PR DESCRIPTION
**Pull request by Hijky**
Background:
Add a testcase to publish API Application

Test Plan
CanPublishAPIApplication
Given create an API application through Control Panel > Object > API Builder
When I click Publish button in edit application wizard
Then the status of the application changed to published

CanPublishAPIApplicationWithChangePublicationStatus
Given create an API application through Control Panel > Object > API Builder
When I click Change Publication Status
Then the status of the application changed to published

CanGetPublishedStatusInResponse
Given create an API application through Control Panel > Object > API Builder
And given publish the API application in edit application wizard
When with getAPIApplicationsPage to retrieve the api application
Then I can get the publish status in response

GetAvailabelButtonsBeforePublish
Given create an API application through Control Panel > Object > API Builder
When I edit the API application
Then Cancel, Save and Publish buttons are visible

GetAvailableButtonsAfterPublish
Given create an API application through Control Panel > Object > API Builder
When I publish the API application in edit application wizard
Then only Cancel and Publish buttons are visible

CanBackToAPIBuilderAfterCancel
Given create an API application through Control Panel > Object > API Builder
And Given I edit the API application
When I click Cancel button
Then it goes back to the API Builder page

CanPromptCancelChangesDialog
Given create an API application through Control Panel > Object > API Builder
And Given I edit the title of the API application
When I click Cancel button
Then I can see the Cancel Changes dialog with message and Continue Without Saving button

CanUpdateTitleAfterRePublish
Given I publish an API application
And Given I save a changed title When I republish the api application
Then I can see updated title in API Builder

CanGetUpdatedTitleInResponseAfterPublish
Given I publish an API application
And Given I edit the title and publish the api application again
When with getAPIApplicationsPage to retrieve the api application
Then I can get the update title in response

Jira ticket:
https://liferay.atlassian.net/browse/LPS-190234

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:sf

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path [hunk 1]
```diff
@@ -20,16 +20,31 @@
 	<td>//div[@class='dropdown-menu show']//button[contains(text(),'Change Publication Status')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>CONTINUE_WITHOUT_SAVING_BUTTON</td>
+	<td>//button[contains(text(),'Continue Without Saving')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>MODAL_CREATE_BUTTON</td>
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path:29]: This is already added in master with a different name, rebase to master to resolve any conflicts.

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+				portlet = "API Builder");
+
+			AssertElementPresent(
+				key_name = "App-test-1",
+				locator1 = "ObjectAdmin#TABLE_LIST_TITLE");
+		}
+	}
+
+	@priority = 3
+	test GetAvailabelButtonsBeforePublish {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase:192]: Typo "available"

**Review by yanan — CHANGES_REQUESTED** [review]
@Hijky , could you please rebase to latest master to make sure there is no conflicts? Thanks

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+		task ("Then the status of the application changed to published") {
+			AssertElementPresent(
+				key_status = "Published",
+				key_title = "App-test",
+				locator1 = "APIBuilder#APPLICATION_STATUS");
+		}
+	}
+
+	@priority = 4
+	test CanUpdateTitleAfterRePublish {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase:161]: This is not same with our original test case "Given I publish an API application". And if you'd like to update, please make sure the test steps are not missing. I mean we miss one step to publish the application before edit the title

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+			var actualValue = JSONUtil.getWithJSONPath(${response}, "$.items..applicationStatus.name");
+
+			TestUtils.assertEquals(
+				actual = ${actualValue},
+				expected = "published");
+		}
+	}
+
+	@priority = 4
+	test CanGetUpdatedTitleInResponseAfterPublish {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase:73]: Same to this case. Missing "Given I publish an API application"

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase (hunk 3 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/PublishAPIApplication.testcase:161]: sorry, in my understanding to create an Application, I will modify it, thanks

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures are not related to PR

**Review by magjed4289 — APPROVED** [review]

**Review by yanan — APPROVED** [review]

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**